### PR TITLE
Feature: Turn off fees and deposits

### PR DESCRIPTION
**Pull request by treygriffith**
## Description
This PR adds support for the new Relayer flag to disable fees and deposits.

It refactors invoice payment into a utility that is used by the OrderStateMachine and FillStateMachine and respects the new flag.

I have yet to write tests, but I wanted to get feedback on the approach and choices that I made before I spend a bunch of time unit testing.

## Todos
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by treygriffith** [comment]
[Turn fees and deposits off](https://trello.com/c/Zx8LN5Nl/857-turn-fees-and-deposits-off)

File: broker-daemon/utils/pay-invoice.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+async function payInvoice (required, paymentRequest, engine, logger, type, publicId) {
```
Comment (martineehrlich) [inline broker-daemon/utils/pay-invoice.js]: jsdoc

File: broker-daemon/models/order.js [hunk 2]
```diff
@@ -153,11 +157,67 @@ class Order {
...
+      throw new Error(`paramsForPlace: orderId is missing.`)
+    }
+
+    if (!outboundSymbol) {
+      throw new Error(`paramsForPlace: outboundSymbol is missing.`)
+    }
+
+    return {
+      feePaymentRequest,
+      feeRequired,
```
Comment (dannypaz) [inline broker-daemon/models/order.js]: passing these fields to the relayer on place or create feels weird. The relayer dictates if the fees/deposits are required and shouldn't have to rely on the broker for this information. Am I misunderstanding this?

Comment (treygriffith) [reply broker-daemon/models/order.js]: We don't pass that to the Relayer. I should probably change the docs for this.

We use that flag in the OrderStateMachine to determine whether or not we pay the fee, but we don't send that to the Relayer.

File: broker-daemon/state-machines/fill-state-machine.js [hunk 3]
```diff
@@ -200,34 +200,25 @@ const FillStateMachine = StateMachine.factory({
...
+        throw new Error(`No engine available for ${outboundSymbol}`)
       }
 
-      const [feeRefundPaymentRequest, depositRefundPaymentRequest] = await Promise.all([
-        outboundEngine.createRefundInvoice(feePaymentRequest),
-        outboundEngine.createRefundInvoice(depositPaymentRequest),
-        outboundEngine.payInvoice(feePaymentRequest),
-        outboundEngine.payInvoice(depositPaymentRequest)
+      const [ feeRefundPaymentRequest, depositRefundPaymentRequest ] = await Promise.all([
+        payInvoice(feeRequired, feePaymentRequest, outboundEngine, this.logger, 'fee', fillId),
```
Comment (dannypaz) [inline broker-daemon/state-machines/fill-state-machine.js]: we should constantize these types (or use the invoice types from the model)

File: broker-daemon/state-machines/fill-state-machine.js [hunk 4]
```diff
@@ -214,9 +214,29 @@ const FillStateMachine = StateMachine.factory({
...
+
+      if (depositRequired) {
+        payDepositInvoice = payInvoice(outboundEngine, depositPaymentRequest)
+      } else {
+        this.logger.debug(`Skipping paying deposit invoice for ${fillId}, not required`)
+      }
+
+      const [
+        feeRefundPaymentRequest,
+        depositRefundPaymentRequest
```
Comment (dannypaz) [inline broker-daemon/state-machines/fill-state-machine.js]: This is a question for below, but will the call to `fillOrder` fail if these are undefined (meaning they are not provided in the request to the relayer)?

File: broker-daemon/utils/pay-invoice.js [hunk 5]
```diff
@@ -8,28 +8,37 @@
...
-  })
-
-  logger.info(`Paid ${type} for ${publicId}`)
-
   return refundPaymentRequest
 }
 
 module.exports = payInvoice
+
+// async function payInvoice (engine, paymentRequest) {
```
Comment (dannypaz) [inline broker-daemon/utils/pay-invoice.js]: forgot to remove the comments

File: broker-daemon/state-machines/fill-state-machine.js (hunk 4 above)
Comment (treygriffith) [reply broker-daemon/state-machines/fill-state-machine.js]: fillOrder won't fail if these are undefined (so long as the fee/deposit was not required)

**Pull request merged**